libs/checkpoint-postgres/langgraph/store/postgres/aio.py

```diff
@@ -260,33 +314,59 @@ async def setup(self) -> None:
         already exist and runs database migrations. It MUST be called directly by the user
         the first time the store is used.
         """
-        async with _ainternal.get_connection(self.conn) as conn:
-            async with conn.cursor() as cur:
-                try:
-                    await cur.execute(
-                        "SELECT v FROM store_migrations ORDER BY v DESC LIMIT 1"
-                    )
-                    row = cast(dict, await cur.fetchone())
-                    if row is None:
-                        version = -1
-                    else:
-                        version = row["v"]
-                except UndefinedTable:
+        async with self._cursor() as cur:
+            try:
+                await cur.execute(
+                    "SELECT v FROM store_migrations ORDER BY v DESC LIMIT 1"
+                )
+                row = await cur.fetchone()
+                if row is None:
                     version = -1
-                    # Create store_migrations table if it doesn't exist
-                    await cur.execute(
-                        """
-                        CREATE TABLE IF NOT EXISTS store_migrations (
-                            v INTEGER PRIMARY KEY
-                        )
-                        """
+                else:
+                    version = row["v"]
+            except UndefinedTable:
+                version = -1
+                await cur.execute(
+                    """
+                    CREATE TABLE IF NOT EXISTS store_migrations (
+                        v INTEGER PRIMARY KEY
                     )
-                for v, migration in enumerate(
-                    self.MIGRATIONS[version + 1 :], start=version + 1
-                ):
-                    await cur.execute(migration)
+                """
+                )
+
+            for v, migration in enumerate(
```

Comment:
Would love critique/suggestions on handling migrations.
Basically, the table needs the emb dimensionality & needs the extension to be installed. It breaks compatibility if we add pgvector as a hard requirement here since you have to install it on the target system.

On the bright side, it's a separate table, so not the worst thing to drop and then re-create if you're early on